sandbox/modules/identity/src/internalClusterTest/java/org/opensearch/identity/UserCrudIT.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.junit.Before;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.test.rest.OpenSearchRestTestCase.entityAsMap;
+
+/**
+ * Tests REST API for users against local cluster
+ */
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class UserCrudIT extends HttpSmokeTestCaseWithIdentity {
+    private final String ENDPOINT;
+
+    public UserCrudIT() {
+        ENDPOINT = ConfigConstants.IDENTITY_REST_API_REQUEST_PREFIX;
+    }
+
+    @Before
+    public void startClusterWithIdentityIndex() throws Exception {
+        startNodesWithIdentityIndex();
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testUsersRestApi() throws Exception {
+
+        String username = "test-create";
+        // Create a user
+        Request request = new Request("PUT", ENDPOINT + "/internalusers/" + username);
+        request.setJsonEntity("{ \"password\" : \"test\" }\n");
+        Response response = getRestClient().performRequest(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 200);
+        Map<String, Object> createResponse = entityAsMap(response);
+        List<Map<String, Object>> usersCreated = (List<Map<String, Object>>) createResponse.get("users");
```

Comment:
Is this how we normally store users when you create one on the dashboard? If not, what do you think about a future test that adds it to the yml or wherever they get added and then checks for it being present?

Reply:
this methods parses the response content. Check out entityAsAMap to see how it parsed the returned json to create a map

Reply:
@scrawfor99 The yml files that are used to seed the security index are never appended to/modified via an API. Security config files (like `internal_users.yml`) must be modified manually and applied using the `securityadmin.sh` tool. 

The API does update the system index (default `.identity_config`) so a test could get the config from the index to ensure the user was added.